client/my-sites/upgrades/index.js

```diff
@@ -289,5 +290,14 @@ module.exports = function() {
 			controller.siteSelection,
 			upgradesController.checkout
 		);
+
+		page(
+			'/start-trial/:domain/:plan_name?',
```

Comment:
I think the convention is to use `:site` over `:domain` for routes.

Moreover I'm pretty sure we should camel case parameters (i.e. `:planName` instead of `:plan_name`).